indexer-app/test/cmr/indexer/test/data/concepts/keyword_util.clj

```diff
@@ -0,0 +1,231 @@
+              "https://acdisc.gesdisc.eosdis.nasa.gov/opendap/aqua_airs_level3/airx3std.006/ "
+              "level3 nasa opendap opendap service service web")
+        (keyword-util/concept-key->keyword-text
+         sample-umm-service-concept :RelatedURL)))
+  (is (= (str "3 analysis and atmosphere atmospheric atmospheric winds cat "
+              "data data analysis and visualization earth earth science "
+              "services engineering geographic geographic information systems "
+              "imagery information microwave microwave imagery radar science "
+              "science cat 3 science term 3 science topic 3 services spectral "
+              "spectral/engineering surface surface winds systems term topic "
+              "visualization winds")
+        (keyword-util/concept-key->keyword-text
+         sample-umm-service-concept :ScienceKeywords))))
+
+(deftest concept-keys->keywords
+  (let [schema-keys [:LongName
+                     :Name
+                     :Version]]
+    (is (= ["1.9" "AIRX3STD" "OPeNDAP Service for AIRS Level-3 retrieval products"]
+           (sort
+            (keyword-util/concept-keys->keywords
+             sample-umm-service-concept schema-keys)))))
+  (let [schema-keys [:LongName
+                     :Name
+                     :Version
+                     :AncillaryKeywords
+                     :ContactGroups
+                     :ContactPersons
+                     :Platforms
+                     :RelatedURL
+                     :ScienceKeywords
+                     :ServiceKeywords
+                     :ServiceOrganizations]]
+    (is (= ["1.9" "A340-600" "ACCESS WEB SERVICE" "AIRX3STD" "ATMOSPHERE" "ATMOSPHERIC WINDS" "AUTHOR" "Airbus A340-600" "Alice" "Bob" "Carol" "CollectionURL" "DATA ANALYSIS AND VISUALIZATION" "DATA ANALYSIS AND VISUALIZATION" "DATA ANALYSIS AND VISUALIZATION" "Data Discovery" "Data Visualization" "EARTH SCIENCE SERVICES" "Eve" "GEOGRAPHIC INFORMATION SYSTEMS" "GET SERVICE" "LDPAAC" "MICROWAVE" "MICROWAVE IMAGERY" "OPeNDAP Service" "OPeNDAP Service for AIRS Level-3 retrieval products" "PUBLISHER" "RADAR" "SCIENCE CAT 3" "SCIENCE CONTACT" "SCIENCE TERM 3" "SCIENCE TOPIC 3" "SERVICE PROVIDER" "SERVICE PROVIDER" "SMWE4B" "SPECTRAL/ENGINEERING" "STATISTICAL APPLICATIONS" "SURFACE WINDS" "Senso-matic Wonder Eye 4B" "TEAM SPOCK" "US GEOLOGICAL SURVEY EARTH RESOURCE OBSERVATION AND SCIENCE (EROS) LANDSAT CUSTOMER SERVICES" "USGS/EROS" "VISUALIZATION/IMAGE PROCESSING" "https://acdisc.gesdisc.eosdis.nasa.gov/opendap/Aqua_AIRS_Level3/AIRX3STD.006/"]
+           (sort
+            (keyword-util/concept-keys->keywords
+             sample-umm-service-concept schema-keys))))))
+
+(deftest concept-keys->keyword-text
+  (let [schema-keys [:LongName
+                     :Name
+                     :Version]]
+    (is (= "1 1.9 3 9 airs airx3std for level opendap opendap service for airs level-3 retrieval products products retrieval service"
```

Comment:
Does order matter when we perform keyword searches? I'm thinking for fields that are a string with multiple words. So someone might want to search for "AIRS level-3" for example. It looks like you are sorting alphabetically the full strings for a field so that if someone searched with multiple words it would be found?

Reply:
That's what I was asking @ygliuvt about last week.

I got more clarity on this last week after poking around a bit in the code and in Elasticsearch. It seems that we get around this by doing two things:
1) we break apart the strings and concatenate them to a large string
2) we also keep the original string and concatenate this too 

The fact that we do both of these explains the occasional duplicates seen in the keyword strings indexed in Elastic and it also removes the loss of adjacency that sorting normally imposes.